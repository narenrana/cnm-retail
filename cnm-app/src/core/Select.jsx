import React from 'react';
import NativeSelect from '@material-ui/core/NativeSelect';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    formControl: {
      margin: theme.spacing(1),
      minWidth: 120,
    },
    selectEmpty: {
      
    },
    root: {
        display: 'flex'
    },
    name: {
        fontSize: '0.8rem',
        padding: '5px 20px',
        fontWeight: '500px'
    }
  }));

export default function Select(props) {
    const {name,value,placeHolder, label,options=[],handleChange}= props;
    const classes = useStyles();
    const onChange=() => {
        if(handleChange) {
            handleChange()
        }
    }
    return (
        <div className={classes.root}>
         <Typography variant="caption"   className={classes.name}>
              {name}
        </Typography>
        <NativeSelect
        className={classes.selectEmpty}
        value={value}
        name="select"
        onChange={onChange}
        inputProps={{ 'aria-label': label}}
      >

        <option value="" disabled>
          {placeHolder}
        </option>
        {
            options.map((option, index)=> <option value={option.value} key={option.value+"-"+index}>{option.label}</option>)
        }
      </NativeSelect>
      </div>
    )
}