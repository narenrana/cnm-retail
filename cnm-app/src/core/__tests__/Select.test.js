/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import Select from "../Select";

describe("Select test", () => {
  it("Select should match snapshot", () => {
    const component = renderer.create(<Select />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
