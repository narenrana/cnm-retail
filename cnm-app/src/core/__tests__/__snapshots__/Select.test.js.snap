// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select test Select should match snapshot 1`] = `
<div
  className="makeStyles-root-3"
>
  <span
    className="MuiTypography-root makeStyles-name-4 MuiTypography-caption"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-selectEmpty-2"
    onClick={[Function]}
  >
    <select
      className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      name="select"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
    >
      <option
        disabled={true}
        value=""
      />
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiNativeSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
